src/steamship/data/embeddings.py

```diff
@@ -195,10 +198,33 @@ def insert_file(
             expect=IndexInsertResponse,
         )
 
+    def _check_input(self, request: IndexInsertRequest, allow_long_records: bool):
```

Comment:
do we want to also check MAX_ACCEPTED_TOKENS (or whatever)?  Or is that better homed plugin-side (as the limits may shift per-provider)?

Reply:
I think that may be better left plugin-side, as tokenization might be different from plugin to plugin anyway, and is somewhat more expensive than a length check.